src/worker/event/kafkaServer.go

```diff
@@ -0,0 +1,119 @@
+package event
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/open-lambda/open-lambda/ol/worker/lambda"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+type KafkaServer struct {
+	lambdaMgr *lambda.LambdaMgr
+}
+
+var kafkaServer *KafkaServer
+var kafkaMu sync.Mutex
+
+func ConvertKafkaMessageToHTTPRequest(message []byte) (*http.Request, error) {
+	// Define the target URL and HTTP method
+
+	url := filepath.Join("/run") // Modify as needed
+	method := "POST"
+
+	// Create a new HTTP request
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(message))
+	if err != nil {
+		return nil, err
+	}
+
+	// Set headers
+	req.Header.Set("User-Agent", "Kafka-Consumer-Client")
+	req.Header.Set("Accept", "*/*")
+	req.Header.Set("Content-Length", fmt.Sprintf("%d", len(message)))
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded") // Assuming Kafka messages are JSON
+
+	return req, nil
+}
+
+func KafkaInit(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can we refactor this function into two?  One for the HTTP side of things, and one for actually starting the goroutine and consuming messages.

Also, it probably makes sense as methods on KafkaServer, no functions that use a global.